app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedViewModel.kt

```diff
@@ -0,0 +1,98 @@
+package org.oppia.android.app.devoptions.markchapterscompleted
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import org.oppia.android.app.fragment.FragmentScope
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.app.model.StorySummary
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.domain.devoptions.ModifyLessonProgressController
+import org.oppia.android.domain.oppialogger.OppiaLogger
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import javax.inject.Inject
+
+/**
+ * [ViewModel] for [MarkChaptersCompletedFragment]. It populates the recyclerview with a list of
+ * [StorySummaryViewModel] which in turn display the story.
+ */
+@FragmentScope
+class MarkChaptersCompletedViewModel @Inject constructor(
+  private val oppiaLogger: OppiaLogger,
+  private val modifyLessonProgressController: ModifyLessonProgressController
+) : ObservableViewModel() {
+
+  private var internalProfileId: Int = -1
```

Comment:
We ought to be passing around a ``ProfileId`` instead of the raw integer. I suggest updating this accordingly.

Reply:
Done.